app/notification.go

```diff
@@ -394,6 +394,9 @@ func (a *App) sendOutOfChannelMentions(sender *model.User, post *model.Post, cha
 	return true, nil
 }
 
+func (a *App) emailNotificationsAllowedForStatus(status string) bool {
```

Comment:
I don't see this as a method, userallowsemails has a lot more logic and it doesn't get a function.
While I applaud the intent of making the condition cleaner and easier to read, it kind of polutes the rest. We can probably achieve the same effect by having a variable which gets the result of this comparison:
```go
statusAllowsNotifications := status.Status != model.STATUS_ONLINE && status.Status != model.STATUS_DND
```

Reply:
@Willyfrog yes.. I thought of adding the above line, but the function is 350 lines huge already. I could see a lot of these one liners adding up also making it hard for understanding and testing.. So extracted that as a separate function. 

Wanted to refactor the `sendNotification` function. but thought to do it in a separate PR.

If you still feel its ok to add the line in the function, I could do the same.  Doing so I really couldn't find an easier way to test something that deep in the function. 

Kindly let me know your thoughts..

Reply:
if the idea is to refactor some of it to remove linecount from the function (which is a fair reason), why not go and extract the whole condition? at a minimum you could add the whole code for userAllowsEmails and you don't even have to change its name as it fits alltogether (if the user is in a certain state he doesn't allowEmails :D )

Reply:
@Willyfrog. You mean we can extract the whole logic into a function. Something like
```
userAllowsEmail := ourNewfunction(.....)

if(userAllowsEmail) {
       a.sendNotificationEmail(notification, profileMap[id], team)
}
```

Kindly let me know if I am understanding it correct.

Reply:
@Willyfrog  changed as we discussed above. Kindly review 👍

Reply:
you forgot to remove the function `emailNotificationsAllowedForStatus` 😄